feat(float): add winborder  option (#31074)
neovim/neovim: runtime/doc/options.txt

	UI-dependent. Works best with RGB colors. 'termguicolors'

						*'winborder'* *'winbd'*
'winborder' 'winbd'	string	(default "")
			global

> 	Configures the border style of floating windows, border values only
> 	support string type which defined in |nvim_open_win()|.

Reviewer: ```suggestion
    Defines the default border style of floating windows. See |nvim_open_win()|
    for valid values.
```

Reviewer: We should probably move the list of border names here, and link to this from |nvim_open_win()|
